### PR TITLE
[Keyboard]add m5 Keyboard.

### PR DESCRIPTION
**Pull request by jonylee1986**
## Description
as title.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/monsgeek/m5/config.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+#define DRIVER_1_LED_TOTAL 61
+#define DRIVER_2_LED_TOTAL 47
+#define RGB_MATRIX_LED_COUNT    (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+
+#define RGB_TRIGGER_ON_KEYDOWN
+#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_KEYPRESSES
+#define RGB_MATRIX_KEYRELEASESs
```
Comment (adophoxia) [inline keyboards/monsgeek/m5/config.h:59]:
```suggestion
#define RGB_MATRIX_KEYRELEASES
```

File: keyboards/monsgeek/m5/config.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#define DRIVER_ADDR_2 0b1110111
+#define I2C1_SDA_PIN B7
+#define I2C1_SCL_PIN B6
+#define I2C1_SCL_PAL_MODE 4
+#define I2C1_OPMODE OPMODE_I2C
+#define I2C1_CLOCK_SPEED 400000 /* 400000 */
+
+#define DRIVER_1_LED_TOTAL 61
+#define DRIVER_2_LED_TOTAL 47
+#define RGB_MATRIX_LED_COUNT    (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
```
Comment (adophoxia) [inline keyboards/monsgeek/m5/config.h:53]:
```suggestion
#define RGB_MATRIX_LED_COUNT 108
```

**Review by adophoxia — CHANGES_REQUESTED** [review]

**Review by adophoxia — APPROVED** [review]

File: keyboards/monsgeek/m5/config.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+/* I2C Config for LED Driver */
+#define DRIVER_COUNT 2
+#define DRIVER_ADDR_1 0b1110100
+#define DRIVER_ADDR_2 0b1110111
+#define I2C1_SDA_PIN B7
+#define I2C1_SCL_PIN B6
+#define I2C1_SCL_PAL_MODE 4
+#define I2C1_OPMODE OPMODE_I2C
+#define I2C1_CLOCK_SPEED 400000 /* 400000 */
```
Comment (waffle87) [inline keyboards/monsgeek/m5/config.h:49]:
```suggestion
#define I2C1_SCL_PAL_MODE 4
```

File: keyboards/monsgeek/m5/info.json [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        "pid": "0x000A",
+        "device_version": "1.0.4",
+        "suspend_wakeup_delay": 400,
+        "force_nkro": true
+    },
+    "processor": "WB32FQ95",
+    "bootloader": "wb32-dfu",
+    "features": {
+        "bootmagic": true,
+        "mousekey": false,
```
Comment (waffle87) [inline keyboards/monsgeek/m5/info.json:16]: Generally preferred to have mousekeys enabled, unless there's a good reason not to (eg. USB endpoints)
```suggestion
        "mousekey": true,
```

File: keyboards/monsgeek/m5/info.json [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": true,
+        "encoder": false,
```
Comment (waffle87) [inline keyboards/monsgeek/m5/info.json:21]:
```suggestion
```

File: keyboards/monsgeek/m5/keymaps/via/rules.mk [hunk 6]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+ENCODER_MAP_ENABLE = no
```
Comment (waffle87) [inline keyboards/monsgeek/m5/keymaps/via/rules.mk:2]:
```suggestion
```

File: keyboards/monsgeek/m5/m5.c [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    writePinLow(LED_CAPS_LOCK_PIN);
+    setPinOutput(LED_WIN_LOCK_PIN); // LED3 Win Lock
+    writePinLow(LED_WIN_LOCK_PIN);
+}
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if (res) {
+        writePin(LED_NUM_LOCK_PIN, led_state.num_lock);
+        writePin(LED_CAPS_LOCK_PIN, led_state.caps_lock);
```
Comment (waffle87) [inline keyboards/monsgeek/m5/m5.c:164]:
```suggestion
void led_init_ports(void) {
    setPinOutput(LED_WIN_LOCK_PIN); // LED3 Win Lock
    writePinLow(LED_WIN_LOCK_PIN);
}

bool led_update_kb(led_t led_state) {
    bool res = led_update_user(led_state);
    if (res) {
```

File: keyboards/monsgeek/m5/readme.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+* Hardware Availability: [monsgeek](https://www.monsgeek.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make monsgeek/m5:default
+
+Flashing example for this keyboard:
+
+    make monsgeek/m5:default:flash
+
```
Comment (waffle87) [inline keyboards/monsgeek/m5/readme.md:16]:
```suggestion

## Bootloader
```

File: keyboards/monsgeek/m5/rules.mk [hunk 9]
```diff
@@ -0,0 +1,6 @@
+# Build Options
+#   change yes to no to disable
+#
+EEPROM_DRIVER = wear_leveling
+WEAR_LEVELING_DRIVER = spi_flash
+
```
Comment (waffle87) [inline keyboards/monsgeek/m5/rules.mk:6]:
```suggestion
EEPROM_DRIVER = wear_leveling
WEAR_LEVELING_DRIVER = spi_flash
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/monsgeek/m5/m5.c [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+    {1,  J_3,   K_3,  L_3},
+    {1,  J_5,   K_5,  L_5},
+    {1,  J_6,   K_6,  L_6},
+
+};
+#endif
+void led_init_ports(void) {
+    setPinOutput(LED_WIN_LOCK_PIN); // LED3 Win Lock
+    writePinLow(LED_WIN_LOCK_PIN);
+}
```
Comment (waffle87) [inline keyboards/monsgeek/m5/m5.c:153]:
```suggestion

void keyboard_pre_init_kb(void) {
    setPinOutput(LED_WIN_LOCK_PIN); // LED3 Win Lock
    writePinLow(LED_WIN_LOCK_PIN);
    keyboard_pre_init_user();
}
```

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**